sopel/modules/reddit.py

```diff
@@ -236,9 +236,58 @@ def comment_info(bot, trigger, match):
     bot.say(message)
 
 
+def subreddit_sorting(bot, trigger, s, sorting):
+    if sorting == 'new':
+        submissions = list(s.new())
+    elif sorting == 'top':
+        submissions = list(s.top())
+    elif sorting == 'hot':
+        submissions = list(s.hot())
+    elif sorting == 'controversial':
+        submissions = list(s.controversial())
+    elif sorting == 'gilded':
+        submissions = list(s.gilded())
+    elif sorting == 'rising':
+        submissions = list(s.rising())
+    elif sorting == 'sticky':
+        try:
+            submissions = [s.sticky()]
+        except prawcore.exceptions.NotFound:
+            bot.say("r/" + s.display_name + " appears to not have a stickied post!")
+            return
+    elif sorting == 'random':
+        submissions = [s.random()] or []
+    else:
+        return
```

Comment:
```suggestion
        raise ValueError("Unknown sorting type '%s'" % sorting)
```

Noisy error because this shouldn't be possible, and probably indicates a code mistake somewhere else.